packages/core/content-manager/admin/src/pages/EditView/components/InputRenderer.tsx

```diff
@@ -23,7 +23,13 @@ import type { EditFieldLayout } from '../../../hooks/useDocumentLayout';
 import type { Schema } from '@strapi/types';
 import type { DistributiveOmit } from 'react-redux';
 
-type InputRendererProps = DistributiveOmit<EditFieldLayout, 'size'>;
+type InputRendererProps = DistributiveOmit<EditFieldLayout, 'size'> & {
+  /**
+   * In the context of content history, deleted fields need to ignore RBAC
+   * @default false
+   */
+  isRBACDisabled?: boolean;
```

Comment:
ℹ️  I’m not crazy about what I have here to handle this case. I left it as is for now so we can discuss first before making big changes. Couple ideas:
- Create our own InputRenderer that is more less the same as this one except with more flexibility around RBAC. So pretty much what I have here but in a file dedicated to content-history. I think this is a good option if we will need a lot more flexibility otherwise I think the current solution ir probably better.
- Change the InputRenderer here to do something similar to our protected pages. Something like an export for InputRenderer (no RBAC checks), then a wrapper component export like ProtectedInputRenderer that handles the RBAC checks before rendering InputRenderer. I don’t know if that abstraction makes sense for this small use case though.

Let me know your thoughts or if you see something I overlooked that could make it even simpler!

Reply:
>I think this is a good option if we will need a lot more flexibility otherwise I think the current solution is probably better.

Agreed, I would say go ahead with see and we'll see if we eventually need to change it. I'm also not a fan of your 2nd idea, more layers create more confusion in my opinion.

Just some feedback in terms of wording though. I don't like that this suggests that RBAC as a whole is disabled, which I don't think is possible. What do you think about something like `shouldIgnorePermissions`?

Reply:
I think you're probably on the money making your own input renderer.

You already need to supply your own relations component so you can not render the input field. The only other bits unique to the CM will be exported as their own components anyway so it should be relatively straight forward for you to accomplish, I don't believe RBAC leaks into the individual fields... _maybe_ components, but I'd be surprised and really question why. It's also just a wrapper around the Form's InputRenderer... 

Thanks for the ping & raising this 🤝 gimme a shout if you need anything else!! 🏄🏻‍♀️